cupy/cusolver.py

```diff
@@ -531,3 +531,138 @@ def gesv(a, b):
     cupy.linalg.util._check_cusolver_dev_info_if_synchronization_allowed(
         getrs, dinfo)
     return b
+
+
+def gels(a, b):
+    """Solves over/well/under-determined linear systems.
+
+    Computes least-square solution to equation ``ax = b` by QR factorization
+    using cusolverDn<t>geqrf().
+
+    Args:
+        a (cupy.ndarray): The matrix with dimension ``(M, N)``.
+        b (cupy.ndarray): The matrix with dimension ``(M)`` or ``(M, K)``.
+
+    Returns:
+        cupy.ndarray:
+            The matrix with dimension ``(N)`` or ``(N, K)``.
+    """
+    if a.ndim != 2:
+        raise ValueError('a.ndim must be 2 (actual: {})'.format(a.ndim))
+    if b.ndim not in (1, 2):
+        raise ValueError('b.ndim must be 1 or 2 (actual: {})'.format(b.ndim))
+    if a.shape[0] != b.shape[0]:
+        raise ValueError('shape mismatch (a: {}, b: {}).'.
+                         format(a.shape, b.shape))
+
+    dtype = numpy.promote_types(a.dtype.char, 'f')
```

Comment:
nit:
```suggestion
    dtype = numpy.promote_types(a.dtype, 'f')
```

By the way, the function may reject int dtypes because the compatibility with numpy does not matter in the submodule.